orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -0,0 +1,217 @@
+    try:
+        model(*args)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_keyword_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetKeywordArguments(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    y = torch.randn(N, D_in, device=device)
+    z = torch.randn(N, D_in, device=device)
+    try:
+        model(x, y, z)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_positional_and_keyword_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetPositionalAndKeywordArguments(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    a = torch.randn(N, D_in, device=device)
+    x = torch.randn(N, D_in, device=device)
+    y = torch.randn(N, D_in, device=device)
+    z = torch.randn(N, D_in, device=device)
+    try:
+        model(a, x, y, z)
+    except Exception as exception:
+        raise exception
+
+def test_model_cuda():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.cuda()
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+def test_model_cpu():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+    model = model.cpu()
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+def test_model_to_device():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.to(device)
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+def test_model_to_tensor():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.to(x)
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
```

Comment:
dont forget the new line at the end :)

Reply:
I think there is a new line already. GitHub doesn't show it well. I think the absence of a new line is indicated by the red circle by GitHub.